ui/app/adapters/pki/issuer.js

```diff
@@ -40,12 +81,25 @@ export default class PkiIssuerAdapter extends ApplicationAdapter {
   }
 
   query(store, type, query) {
-    return this.ajax(this.urlForQuery(query.backend), 'GET', this.optionsForQuery());
+    const { backend, shouldShowIssuerMetaData } = query;
+    const url = this.urlForQuery(backend);
+
+    return this.ajax(url, 'GET', this.optionsForQuery()).then(async (res) => {
+      // To show issuer meta data tags, we have a flag called shouldShowIssuerMetaData and only want to
+      // grab each issuer data only if there are less than 10 issuers to avoid making too many requests
+      if (shouldShowIssuerMetaData && res.data.keys.length <= 10) {
+        const issuerMetaData = this.getIssuerMetaData(store, type, query, res);
```

Comment:
It looks like you are missing an `await` here. I'm also wondering about error handling for `getIssuerMetaData`. It looks like there are multiple async requests being made so do we want to just pass along whatever the specific error is or maybe add a catch with a specific error message that getting the issuer meta data failed?